http/src/main/java/com/mx/path/connect/http/HttpClientFilter.java

```diff
@@ -157,6 +161,15 @@ public final void execute(Request request, Response response) {
           } finally {
             response.finish();
           }
+        } catch (ConnectTimeoutException e) {
+          httpResponse.setStatus(HttpStatus.valueOf(HTTP_STATUS_EXTERNAL_TIMEOUT));
```

Comment:
Setting the status here is meaningless if you are just going to immediately raise an exception.

Reply:
I think you may be trying to solve this too close to the call. Have you looked at the request filters? That may be a more appropriate place to solve for this. https://github.com/mxenabled/path-core/blob/d286c4221330adc010e408c3edf4c2d44def39f8/gateway/src/main/java/com/mx/path/gateway/connect/filter/ErrorHandlerFilter.java#L52

Reply:
If this is the right place (which I'm not sure about), then we should be raising HttpClientConnectionExceptions, and setting the status in those. There is no reason for the extra constant. We should use the the PathResponseStatus enum.